wasi_snapshot_preview1/wasi.go

```diff
@@ -1382,34 +1382,34 @@ func (a *wasi) SchedYield(m api.Module) Errno {
 //
 // Note: importRandomGet shows this signature in the WebAssembly 1.0 (20191205) Text Format.
 // See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-random_getbuf-pointeru8-bufLen-size---errno
-func (a *wasi) RandomGet(ctx context.Context, m api.Module, buf uint32, bufLen uint32) (errno Errno) {
-	randomBytes := make([]byte, bufLen)
```

Comment:
cc @clarkmcc here's the interesting part, which is you don't need to make a separate buffer to write to memory. Instead of make(buf), xx.Read(buf), mem.Write. We can do buf = mem.Read then xx.Read(buf)